xbmc/rendering/dx/RenderSystemDX.cpp

```diff
@@ -425,8 +425,8 @@ void CRenderSystemDX::DeleteDevice()
     m_pGUIShader->End();
 
   // tell any shared resources
-  for (std::vector<ID3DResource *>::iterator i = m_resources.begin(); i != m_resources.end(); ++i)
-    (*i)->OnDestroyDevice();
+  for (std::vector<IDispResource *>::iterator i = m_resources.begin(); i != m_resources.end(); ++i)
+    (*i)->OnDestroyDisplay();
```

Comment:
this wrong. All ID3DResource still await OnDestroyDevice to destroy any device related resources like a textures, shaders and so one.